spec/models/callback_modifier_spec.rb

```diff
@@ -23,6 +23,19 @@
             modifier.test_destroy
             expect(modifier.versions.last.reify).to be_flagged_deleted
           end
+
+          if active_record_gem_version.release >= Gem::Version.new("5")
+            context "when ActiveRecord::Base.belongs_to_required_by_default is enabled" do
+              it "should display a deprecation warning" do
+                allow(::ActiveRecord::Base).to receive(:belongs_to_required_by_default).
+                  and_return(true)
+                expect(::ActiveSupport::Deprecation).to receive(:warn).
+                  with(/paper_trail_on_destroy(:after) is incompatible with ActiveRecord/)
```

Comment:
I think the real issue is that `paper_trail_on_destroy` is a class method that is called as soon as the `AfterDestroyModifier` class is loaded, so the warning is happening before the example is run in this spec.